sdk/identity/Azure.Identity/src/DateTimeHelpers.netfx.cs

```diff
@@ -0,0 +1,77 @@
+﻿﻿﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Azure.Identity
+{
+    // Implementation is taken from https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.DateTime.netfx.cs
+    internal static partial class DateTimeHelpers
+    {
+        private static TimeSync _timeSync = new TimeSync();
+
+        // sync DateTime and Stopwatch ticks every 2 hours
+#pragma warning disable CA1823 // suppress unused field warning, as it's used to keep the timer alive
+        private static readonly Timer _syncTimeUpdater = InitializeSyncTimer();
+#pragma warning restore CA182
+
+        private static TimerCallback _sync = s =>
+        {
+            // wait for DateTime.UtcNow update to the next granular value
+            Thread.Sleep(1);
+            Interlocked.Exchange(ref _timeSync, new TimeSync());
+        };
+
+        /// <summary>
+        /// Returns high resolution (1 DateTime tick) current UTC DateTime.
+        /// </summary>
+        private static DateTime GetUtcNowImplementation()
+        {
+            // DateTime.UtcNow accuracy on .NET Framework is ~16ms, this method
```

Comment:
Do we care about 16ms error?

Reply:
`System.Diagnostics.Activity` will have wrong `Duration` if we don't use the same precision they use:
https://github.com/dotnet/runtime/blob/21ea59f90044f4f662117fc43022f07eafb485cd/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L521

Not sure how critical is that, but IMO it is good to have consistent behavior of `UtcNow` across different frameworks.

Reply:
Is this done in `Azure.Core` already? If not I agree that it's adds a good bit of complexity for something which doesn't really impact users.

Reply:
Ok, reverting it.

Reply:
Reverted